### PR TITLE
pass DOCKER_EXPERIMENTAL into build images

### LINKED ISSUE #14896: Experimental build isn't experimental
Something seems to be wrong with the experimental build:

```
$ curl -sSL https://experimental.docker.com/ | sh
...
+ sh -c docker version
Client:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   a031b64
 Built:        Wed Jul 15 05:21:52 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.0-dev
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   a031b64
 Built:        Wed Jul 15 05:21:52 UTC 2015
 OS/Arch:      linux/amd64

If you would like to use Docker as a non-root user, you should now consider
adding your user to the "docker" group with something like:

  sudo usermod -aG docker your-user

Remember that you will have to log out and back in for this to take effect!

$ docker -v
Docker version 1.8.0-dev, build a031b64
```

Note that this is *not* an experimental build, but a stable build of master.

### PR DESCRIPTION
**Pull request by jessfraz**
fixes #14896

ping @tianon its early idk if this is the best route, esp when the var is empty...

### CODE REVIEW

**Comment by icecrime** [comment]
LGTM

**Comment by tianon** [comment]
Won't this just break if the var is empty?

**Comment by jfrazelle** [comment]
ya im trying it

**Comment by jfrazelle** [comment]
i think if we put "" then it will be fine

**Comment by tianon** [comment]
Why not just conditionally add the line with an `if` to be safe?
```bash
		cat > "$DEST/$version/Dockerfile.build" <<-EOH
			FROM $image
			WORKDIR /usr/src/docker
			COPY . /usr/src/docker
		EOH
		if [ "$DOCKER_EXPERIMENTAL" ]; then
			echo 'ENV DOCKER_EXPERIMENTAL 1' >> "$DEST/$version/Dockerfile.build"
		fi
		cat >> "$DEST/$version/Dockerfile.build" <<-EOF
			RUN ln -sfv hack/make/.build-deb debian
			RUN { echo '$debSource (${debVersion}-0~${suite}) $suite; urgency=low'; echo; echo '  * Version: $VERSION'; echo; echo " -- $debMaintainer  $debDate"; } > debian/changelog && cat >&2 debian/changelog
			RUN dpkg-buildpackage -uc -us
		EOF
```

**Comment by tianon** [comment]
Putting the quotes in like that will likely just make it always truthy. :confused:

**Comment by jfrazelle** [comment]
ah yeah ok cool i just felt gross braking the cat EOF like that but i like it

**Comment by jfrazelle** [comment]
updated

File: hack/make/build-deb [hunk 1]
```diff
@@ -52,6 +52,11 @@ set -e
 			FROM $image
 			WORKDIR /usr/src/docker
 			COPY . /usr/src/docker
+		EOF
+        if [ "$DOCKER_EXPERIMENTAL" ]; then
```
Comment (jfrazelle) [inline hack/make/build-deb]: ew tabs da fak

**Comment by tianon** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Comment by jfrazelle** [comment]
ok rebuilding and pushing experimental, should be like an hour

**Pull request merged**